docs/how-to/upgrade/major-version-upgrade.md

```diff
@@ -0,0 +1,183 @@
+
+Then, integrate the new cluster with S3-integrator. If you're deploying your new cluster in the same model, it should be as easy as {command}`juju integrate mongodb-seven s3-integrator`. See {ref}`configure-s3-aws` for more information.
+
+Replicate all the passwords from your initial application to the new one:
+
+```shell
+for username in backup orchestrator monitor logrotate; do
+    password=$(juju run <app>/leader get-password username=$username);
+    juju run <app>-seven/leader set-password username=$username password=$password;
+done
+```
+
+## Back up MongoDB 6 cluster
+
+Now, create a backup of the initial cluster, and note the `<backup-id>`. Wait for it to finish.
+
+```shell
+juju run <app>/leader create-backup
+```
+
+<!--TODO: Clarify which cluster we are logging into here - the old 6 or new 7?-->
+{ref}`Log into your cluster <access-a-replica-set>` and set the feature compatibility version to `6.0`. This will allow a safe restore to the new cluster.
+
+```javascript
+db.adminCommand(
+    {
+        setFeatureCompatibilityVersion: "6.0",
+        confirm: true,
+    }
+)
+```
+
+## Restore backup on MongoDB 7 cluster
+
+Restore the MongoDB 6 backup on the MongoDB 7 cluster:
+
+```shell
+juju run <app>-seven/leader restore backup-id=<backup-id>
+```
+
+This is the time to ensure stability of your deployment. When you are sure you want to continue, run:
+
+```javascript
+db.adminCommand(
+    {
+        setFeatureCompatibilityVersion: "7.0",
+        confirm: true,
+    }
+)
+```
+
+Your cluster is now running MongoDB 7, and you can proceed to the next part of this documentation for upgrading to MongoDB 8.
+
+## Deploy a MongoDB 8 cluster
+
+These steps are pretty similar to the previous ones, but we will go through it again.
+
+{ref}`Deploy <how-to-deploy>` a MongoDB 8 cluster from the Charmhub channel `8/stable` with the same topology as your initial cluster. For example, if your cluster has 1 config-server and three shards, your `8` cluster should also have 1 config-server and three shards.
+
+Then, integrate that cluster with S3-integrator. If you're deploying your new cluster in the same model, it should be as easy as {command}`juju integrate mongodb-seven s3-integrator`. See {ref}`configure-s3-aws` for more information.
+
+Replicate all the passwords from your initial application to the new one:
+
+```shell
+for username in backup orchestrator monitor logrotate; do
+    password=$(juju run <app>/leader get-password username=$username);
+    juju run <app>-eight/leader set-password username=$username password=$password;
+done
+```
+
+## Back up MongoDB 7 cluster
+
+Now, create a backup of the MongoDB 7 cluster, and note the `<backup-id>`. Wait for it to finish.
+
+```shell
+juju run <app>-seven/leader create-backup
+```
+
+{ref}`Log into your cluster <access-a-replica-set>` and set the feature compatibility version to `6.0`. This will allow a safe restore to the new cluster.
```

Comment:
```suggestion
{ref}`Log into your cluster <access-a-replica-set>` and set the feature compatibility version to `7.0`. This will allow a safe restore to the new cluster.
```